doc/man/man3/MasterActions.3
.TH "MasterActions" 3 "Fri Jun 12 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
MasterActions \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <MasterActions\&.h>\fP
.PP
Inherits QWidget\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBMasterActions\fP (QWidget *parent=0)"
.br
.ti -1c
.RI "void \fBinitActions\fP ()"
.br
.ti -1c
.RI "void \fBsetNorthTabWidgetPtr\fP (QTabWidget *\fBnorthTabWidgetPtr\fP)"
.br
.ti -1c
.RI "QTabWidget * \fBgetNorthTabWidgetPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetSouthTabWidgetPtr\fP (QTabWidget *\fBsouthTabWidgetPtr\fP)"
.br
.ti -1c
.RI "QTabWidget * \fBgetSouthTabWidgetPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetHighlighter\fP (\fBHighlighter\fP *\fBhighlighter\fP)"
.br
.ti -1c
.RI "\fBHighlighter\fP * \fBgetHighlighter\fP ()"
.br
.ti -1c
.RI "void \fBsetNewFileGuiPtr\fP (\fBNewFileGui\fP *\fBnewFileGuiPtr\fP)"
.br
.ti -1c
.RI "\fBNewFileGui\fP * \fBgetNewFileGuiPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetNewProjectGuiPtr\fP (\fBNewProjectGui\fP *\fBnewProjectGuiPtr\fP)"
.br
.ti -1c
.RI "\fBNewProjectGui\fP * \fBgetNewProjectGuiPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetTerminalPtr\fP (\fBTerminal\fP *\fBterminalPtr\fP)"
.br
.ti -1c
.RI "\fBTerminal\fP * \fBgetTerminalPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetRunGuiPtr\fP (\fBRunGui\fP *\fBrunGuiPtr\fP)"
.br
.ti -1c
.RI "\fBRunGui\fP * \fBgetRunGuiPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetOpenProjectGuiPtr\fP (\fBOpenProjectGui\fP *\fBopenProjectGuiPtr\fP)"
.br
.ti -1c
.RI "\fBOpenProjectGui\fP * \fBgetOpenProjectGuiPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetWestWidgetPtr\fP (\fBFileTreeGui\fP *\fBwestWidgetPtr\fP)"
.br
.ti -1c
.RI "\fBFileTreeGui\fP * \fBgetWestWidgetPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetEastWidgetPtr\fP (\fBNavigatorGui\fP *\fBeastWidgetPtr\fP)"
.br
.ti -1c
.RI "\fBNavigatorGui\fP * \fBgetEastWidgetPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetBuildPtr\fP (\fBBuild\fP *\fBbuildPtr\fP)"
.br
.ti -1c
.RI "\fBBuild\fP * \fBgetBuildPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetNewFileActionPtr\fP (QAction *\fBnewFileActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetNewFileActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetNewTerminalActionPtr\fP (QAction *\fBnewTerminalActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetNewTerminalActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetNewProjectActionPtr\fP (QAction *\fBnewProjectActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetNewProjectActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetOpenProjectActionPtr\fP (QAction *\fBopenProjectActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetOpenProjectActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetSaveAllActionPtr\fP (QAction *\fBsaveAllActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetSaveAllActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetUndoActionPtr\fP (QAction *\fBundoActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetUndoActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetRedoActionPtr\fP (QAction *\fBredoActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetRedoActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetSetProjectConfigActionPtr\fP (QAction *\fBsetProjectConfigActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetSetProjectConfigActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetBuildActionPtr\fP (QAction *setBuildActionPtr)"
.br
.ti -1c
.RI "QAction * \fBgetBuildActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetCleanAndBuildActionPtr\fP (QAction *\fBcleanAndBuildActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetCleanAndBuildActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetConfigForRunActionPtr\fP (QAction *\fBconfigForRunActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetConfigForRunActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetRunActionPtr\fP (QAction *\fBrunActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetRunActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetDebugActionPtr\fP (QAction *\fBdebugActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetDebugActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetGitActionPtr\fP (QAction *profileProjActionPtr)"
.br
.ti -1c
.RI "QAction * \fBgetGitActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetOutputSouthActionPtr\fP (QAction *\fBoutputSouthActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetOutputSouthActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetTerminalSouthActionPtr\fP (QAction *\fBterminalSouthActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetTerminalSouthActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetNavEastActionPtr\fP (QAction *navEastActionptr)"
.br
.ti -1c
.RI "QAction * \fBgetNavEastActionPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetProjectWestActionPtr\fP (QAction *\fBprojectWestActionPtr\fP)"
.br
.ti -1c
.RI "QAction * \fBgetProjectWestActionPtr\fP ()"
.br
.ti -1c
.RI "template<class X > void \fBconnectToNewFileAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToNewTerminalAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToNewProjectAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToOpenProjectAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToSaveAllAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToUndoAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToRedoAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToSetProjectConfigAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToBuildAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToCleanAndBuildAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToConfigForRunAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToRunAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToDebugAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToGitAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToOutputSouthAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToTerminalSouthAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToNavEastAction\fP (X *component)"
.br
.ti -1c
.RI "template<class X > void \fBconnectToProjectWestAction\fP (X *component)"
.br
.ti -1c
.RI "\fB~MasterActions\fP ()"
.br
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBhandleNewFileActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleNewTerminalActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleNewProjectActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleOpenProjectActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleSaveAllActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleUndoActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleRedoActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleSetProjectConfigActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleBuildActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleCleanAndBuildActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleConfigForRunActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleRunActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleDebugActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleGitActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleOutputSouthActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleTerminalSouthActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleNavEastActionSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleProjectWestActionSlot\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "QAction * \fBnewFileActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBnewTerminalActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBnewProjectActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBopenProjectActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBsaveAllActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBundoActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBredoActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBsetProjectConfigActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBbuildActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBcleanAndBuildActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBconfigForRunActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBrunActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBdebugActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBgitActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBoutputSouthActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBterminalSouthActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBnavEastActionPtr\fP"
.br
.ti -1c
.RI "QAction * \fBprojectWestActionPtr\fP"
.br
.ti -1c
.RI "QTabWidget * \fBnorthTabWidgetPtr\fP"
.br
.ti -1c
.RI "QTabWidget * \fBsouthTabWidgetPtr\fP"
.br
.ti -1c
.RI "\fBHighlighter\fP * \fBhighlighter\fP"
.br
.ti -1c
.RI "\fBNewFileGui\fP * \fBnewFileGuiPtr\fP"
.br
.ti -1c
.RI "\fBNewProjectGui\fP * \fBnewProjectGuiPtr\fP"
.br
.ti -1c
.RI "\fBTerminal\fP * \fBterminalPtr\fP"
.br
.ti -1c
.RI "\fBRunGui\fP * \fBrunGuiPtr\fP"
.br
.ti -1c
.RI "\fBOpenProjectGui\fP * \fBopenProjectGuiPtr\fP"
.br
.ti -1c
.RI "\fBFileTreeGui\fP * \fBwestWidgetPtr\fP"
.br
.ti -1c
.RI "\fBNavigatorGui\fP * \fBeastWidgetPtr\fP"
.br
.ti -1c
.RI "\fBBuild\fP * \fBbuildPtr\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 37 of file MasterActions\&.h\&.

src/MasterActions.cpp
#include "MasterActions.h"
#include "OutputGui.h"


MasterActions::MasterActions(QWidget* parent) : QWidget(parent)
{
    runGuiPtr = new RunGui();
    openProjectGuiPtr = new OpenProjectGui();
    buildPtr = new Build();
    
    // North
    newFileActionPtr = new QAction(QIcon("images/newFile.jpg"), tr("&New File"), this);
    newFileActionPtr->setShortcut(QKeySequence::New);
    newFileActionPtr->setStatusTip("New File");
    connect(newFileActionPtr, SIGNAL(triggered() ), this, SLOT(handleNewFileActionSlot() ) );
    
    newTerminalActionPtr = new QAction(QIcon("images/terminal.jpg"), tr("&New Terminal"), this);
    newTerminalActionPtr->setShortcut(QKeySequence::New);
    newTerminalActionPtr->setStatusTip("New Terminal");
    connect(newTerminalActionPtr, SIGNAL(triggered() ), this, SLOT(handleNewTerminalActionSlot() ) );
    
    newProjectActionPtr = new QAction(QIcon("images/newProject.jpg"), tr("&New Project"), this);
    newProjectActionPtr->setShortcut(QKeySequence::New);
    newProjectActionPtr->setStatusTip("New Project");
    connect(newProjectActionPtr, SIGNAL(triggered() ), this, SLOT(handleNewProjectActionSlot() ) );
    
    openProjectActionPtr = new QAction(QIcon("images/openProject.jpg"), tr("&Open Project"), this);
    openProjectActionPtr->setShortcut(QKeySequence::New);
    openProjectActionPtr->setStatusTip("Open Existing Project");
    connect(openProjectActionPtr, SIGNAL(triggered() ), this, SLOT(handleOpenProjectActionSlot() ) );
    
    saveAllActionPtr = new QAction(QIcon("images/save.jpg"), tr("&Save All"), this);
    saveAllActionPtr->setShortcut(QKeySequence::New);
    saveAllActionPtr->setStatusTip("Save All");
    connect(saveAllActionPtr, SIGNAL(triggered() ), this, SLOT(handleSaveAllActionSlot() ) );
    
    undoActionPtr = new QAction(QIcon("images/undo.jpg"), tr("&Undo"), this);
    undoActionPtr->setShortcut(QKeySequence::New);
    undoActionPtr->setStatusTip("Undo");
    connect(undoActionPtr, SIGNAL(triggered() ), this, SLOT(handleUndoActionSlot() ) );
    
    redoActionPtr = new QAction(QIcon("images/redo.jpg"), tr("&Redo"), this);
    redoActionPtr->setShortcut(QKeySequence::New);
    redoActionPtr->setStatusTip("Redo");
    connect(redoActionPtr, SIGNAL(triggered() ), this, SLOT(handleRedoActionSlot() ) );

    setProjectConfigActionPtr = new QAction(QIcon("images/generate.jpg"), tr("&set Project Configuration"), this);
    setProjectConfigActionPtr->setShortcut(QKeySequence::New);
    setProjectConfigActionPtr->setStatusTip("Configure");
    connect(setProjectConfigActionPtr, SIGNAL(triggered() ), this, SLOT(handleSetProjectConfigActionSlot() ) );
    
    buildActionPtr = new QAction(QIcon("images/build.jpg"), tr("&Build Project"), this);
    buildActionPtr->setShortcut(QKeySequence::New);
    buildActionPtr->setStatusTip("Build");
    connect(buildActionPtr, SIGNAL(triggered() ), this, SLOT(handleBuildActionSlot() ) );
    
    cleanAndBuildActionPtr = new QAction(QIcon("images/cleanAndBuild.jpg"), tr("&Clean and Build Project"), this);
    cleanAndBuildActionPtr->setShortcut(QKeySequence::New);
    cleanAndBuildActionPtr->setStatusTip("Clean and Build");
    connect(cleanAndBuildActionPtr, SIGNAL(triggered() ), this, SLOT(handleCleanAndBuildActionSlot() ) );
    
    configForRunActionPtr = new QAction(QIcon("images/configForRun.jpg"), tr("&Configure for Execution"), this);
    configForRunActionPtr->setShortcut(QKeySequence::New);
    configForRunActionPtr->setStatusTip("Configure for execution");
    connect(configForRunActionPtr, SIGNAL(triggered() ), this, SLOT(handleConfigForRunActionSlot() ) );
    
    runActionPtr = new QAction(QIcon("images/run.jpg"), tr("&Run Project"), this);
    runActionPtr->setShortcut(QKeySequence::New);
    runActionPtr->setStatusTip("Run Project");
    connect(runActionPtr, SIGNAL(triggered() ), this, SLOT(handleRunActionSlot() ) );
    
    debugActionPtr = new QAction(QIcon("images/debug.jpg"), tr("&Debug Project"), this);
    debugActionPtr->setShortcut(QKeySequence::New);
    debugActionPtr->setStatusTip("Debug");
    connect(debugActionPtr, SIGNAL(triggered() ), this, SLOT(handleDebugActionSlot() ) );
    
    gitActionPtr = new QAction(QIcon("images/github.jpeg"), tr("&Git/GitHub"), this);
    gitActionPtr->setShortcut(QKeySequence::New);
    gitActionPtr->setStatusTip("Git");
    connect(gitActionPtr, SIGNAL(triggered() ), this, SLOT(handleGitActionSlot() ) );

    // South
    outputSouthActionPtr = new QAction(QIcon("images/output.jpg"), tr("&Output"), this);
    outputSouthActionPtr->setShortcut(QKeySequence::New);
    outputSouthActionPtr->setStatusTip("Output");
    connect(outputSouthActionPtr, SIGNAL(triggered() ), this, SLOT(handleOutputSouthActionSlot() ) );
    
    terminalSouthActionPtr = new QAction(QIcon("images/terminal.jpg"), tr("&Output"), this);
    terminalSouthActionPtr->setShortcut(QKeySequence::New);
    terminalSouthActionPtr->setStatusTip("Terminal");
    connect(terminalSouthActionPtr, SIGNAL(triggered() ), this, SLOT(handleTerminalSouthActionSlot() ) );
    
    // East
    navEastActionPtr = new QAction(QIcon("images/navigator.jpg"), tr("&Output"), this);
    navEastActionPtr->setShortcut(QKeySequence::New);
    navEastActionPtr->setStatusTip("Navigator");
    connect(navEastActionPtr, SIGNAL(triggered() ), this, SLOT(handleNavEastActionSlot() ) );
    
    // West
    projectWestActionPtr = new QAction(QIcon("images/tree.jpg"), tr("&Output"), this);
    projectWestActionPtr->setShortcut(QKeySequence::New);
    projectWestActionPtr->setStatusTip("Project Tree");
    connect(projectWestActionPtr, SIGNAL(triggered() ), this, SLOT(handleProjectWestActionSlot() ) );
    
    
}


void MasterActions::handleNewFileActionSlot()
{
    newFileGuiPtr = new NewFileGui();
    newFileGuiPtr->setNorthTabWidgetPtr(northTabWidgetPtr);
    newFileGuiPtr->show();
}

void MasterActions::handleNewTerminalActionSlot()
{
    cout << "New Terminal activated" << endl;
    
    Terminal* term = new Terminal();
    northTabWidgetPtr->addTab(term, tr("RosTerm"));
    northTabWidgetPtr->setCurrentIndex( (northTabWidgetPtr->children().size() - 1) );
    term->start();
}


void MasterActions::handleNewProjectActionSlot()
{
    newProjectGuiPtr = new NewProjectGui();
    newProjectGuiPtr->show();
}


void MasterActions::handleOpenProjectActionSlot()
{
    openProjectGuiPtr->show();
}


void MasterActions::handleSaveAllActionSlot()
{
    SaveAll::save();
}


void MasterActions::handleUndoActionSlot()
{
    ;
}


void MasterActions::handleRedoActionSlot()
{
    ;
}


void MasterActions::handleSetProjectConfigActionSlot()
{
    ;
}


void MasterActions::handleBuildActionSlot()
{
    RProcess* rprocess = new RProcess();
    rprocess->setOutputLocTePtr(southTabWidgetPtr->getOutputTePtr() );
    rprocess->start/*Detached*/("catkin_make"); // pass output tab to rprocess?
    cout << "# of child tabs: " << southTabWidgetPtr->children().size() << endl;
    southTabWidgetPtr->getTabWidget()->setTabText(0, "Overriding!");
    //southTabWidgetPtr->dumpToGui(new QString("compiling..."));
}


void MasterActions::handleCleanAndBuildActionSlot()
{
    cout << "At MasterActions::handleCleanAndBuildActionSlot(), assuming that ros workspace"
         << "\nis two directories above project root dir" << endl;
    QProcess* build = new QProcess();
    build->setWorkingDirectory("./../../" + *openProjectGuiPtr->getProjectStrPtr() );
    build->start("catkin_make");
}


void MasterActions::handleConfigForRunActionSlot()
{
    runGuiPtr->reset();
    runGuiPtr->show();
}


void MasterActions::handleRunActionSlot()
{
    RProcess* rprocess = new RProcess();
    rprocess->setOutputLocTePtr(southTabWidgetPtr->getOutputTePtr() );
    QStringList* tmpStrLstPtr = new QStringList();
    cout << "!!!: " << runGuiPtr->getRunPage_2Ptr()->getPkgStrPtr().toStdString();
    cout << "!!!: " << runGuiPtr->getRunPage_2Ptr()->getFileStrPtr().toStdString() << endl;
    tmpStrLstPtr->push_back(runGuiPtr->getRunPage_2Ptr()->getPkgStrPtr() );
    tmpStrLstPtr->push_back(runGuiPtr->getRunPage_2Ptr()->getFileStrPtr() );
    rprocess->start/*Detached*/("roslaunch", *tmpStrLstPtr);
}


void MasterActions::handleDebugActionSlot()
{
    ;
}


void MasterActions::handleGitActionSlot()
{
    ;
}


// South
void MasterActions::handleOutputSouthActionSlot()
{
    if(getSouthTabWidgetPtr()->getTabWidget()->isVisible() )
    {
        getSouthTabWidgetPtr()->getTabWidget()->hide();
    }
    else
    {
        getSouthTabWidgetPtr()->getTabWidget()->show();
    }
}


void MasterActions::handleTerminalSouthActionSlot()
{
    cout << "New Terminal activated" << endl;
    
    Terminal* term = new Terminal();
    southTabWidgetPtr->getTabWidget()->addTab(term, tr("RosTerm"));
    
    term->start();
}


// East
void MasterActions::handleNavEastActionSlot()
{
    if(getEastWidgetPtr()->isVisible() )
    {
        getEastWidgetPtr()->hide();
    }
    else
    {
        getEastWidgetPtr()->show();
    }
}


// West
void MasterActions::handleProjectWestActionSlot()
{
    if(getWestWidgetPtr()->isVisible() )
    {
        getWestWidgetPtr()->hide();
    }
    else
    {
        getWestWidgetPtr()->show();
    }
}


void MasterActions::initActions()
{
    ;
}


void MasterActions::setNorthTabWidgetPtr(QTabWidget* northTabWidgetPtr)
{
    this->northTabWidgetPtr = northTabWidgetPtr;
}


QTabWidget* MasterActions::getNorthTabWidgetPtr()
{
    return northTabWidgetPtr;
}


void MasterActions::setSouthTabWidgetPtr(OutputGui* /*QTabWidget**/ southTabWidgetPtr)
{
    this->southTabWidgetPtr = southTabWidgetPtr;
}


OutputGui* /*QTabWidget**/ MasterActions::getSouthTabWidgetPtr()
{
    return southTabWidgetPtr;
}


void MasterActions::setHighlighter(Highlighter* highlighter)
{
    this->highlighter = highlighter;
}


Highlighter* MasterActions::getHighlighter()
{
    return highlighter;
}


void MasterActions::setNewFileGuiPtr(NewFileGui* newFileGuiPtr)
{
    this->newFileGuiPtr = newFileGuiPtr;
}


NewFileGui* MasterActions::getNewFileGuiPtr()
{
    return newFileGuiPtr;
}


void MasterActions::setNewProjectGuiPtr(NewProjectGui* newProjectGuiPtr)
{
    this->newProjectGuiPtr = newProjectGuiPtr;
}


NewProjectGui* MasterActions::getNewProjectGuiPtr()
{
    return newProjectGuiPtr;
}


void MasterActions::setTerminalPtr(Terminal* terminalPtr)
{
    this->terminalPtr = terminalPtr;
}


Terminal* MasterActions::getTerminalPtr()
{
    return terminalPtr;
}


void MasterActions::setRunGuiPtr(RunGui* runGuiPtr)
{
    this->runGuiPtr = runGuiPtr;
}


RunGui* MasterActions::getRunGuiPtr()
{
    return runGuiPtr;
}


void MasterActions::setOpenProjectGuiPtr(OpenProjectGui* openProjectGuiPtr)
{
    this->openProjectGuiPtr = openProjectGuiPtr;
}


OpenProjectGui* MasterActions::getOpenProjectGuiPtr()
{
    return openProjectGuiPtr;
}


void MasterActions::setWestWidgetPtr(FileTreeGui* fileTreeGuiPtr)
{
    this->westWidgetPtr = fileTreeGuiPtr;
    fileTreeGuiPtr->setNorthTabWidgetPtr(getNorthTabWidgetPtr() );
    cout << cct::yellow("move line at: MasterActions::setWestWidgetPtr(...)") << endl;
}


FileTreeGui* MasterActions::getWestWidgetPtr()
{
    return westWidgetPtr;
}


void MasterActions::setEastWidgetPtr(NavigatorGui* eastWidgetPtr)
{
    this->eastWidgetPtr = eastWidgetPtr;
}


NavigatorGui* MasterActions::getEastWidgetPtr()
{
    return eastWidgetPtr;
}


void MasterActions::setBuildPtr(Build* buildPtr)
{
    this->buildPtr = buildPtr;
}


Build* MasterActions::getBuildPtr()
{
    return buildPtr;
}


// Accessors and Mutators for actions
void MasterActions::setNewFileActionPtr(QAction* newFileActionPtr)
{
    this->newFileActionPtr = newFileActionPtr;
}


QAction* MasterActions::getNewFileActionPtr()
{
    return newFileActionPtr;
}


void MasterActions::setNewTerminalActionPtr(QAction* newTerminalActionPtr)
{
    this->newTerminalActionPtr = newTerminalActionPtr;
}


QAction* MasterActions::getNewTerminalActionPtr()
{
    return newTerminalActionPtr;
}


void MasterActions::setNewProjectActionPtr(QAction* newProjectActionPtr)
{
    this->newProjectActionPtr = newProjectActionPtr;
}


QAction* MasterActions::getNewProjectActionPtr()
{
    return newProjectActionPtr;
}


void MasterActions::setOpenProjectActionPtr(QAction* openProjectActionPtr)
{
    this->openProjectActionPtr = openProjectActionPtr;
}


QAction* MasterActions::getOpenProjectActionPtr()
{
    return openProjectActionPtr;
}


void MasterActions::setSaveAllActionPtr(QAction* saveAllActionPtr)
{
    this->saveAllActionPtr = saveAllActionPtr;
}


QAction* MasterActions::getSaveAllActionPtr()
{
    return saveAllActionPtr;
}


void MasterActions::setUndoActionPtr(QAction* undoActionPtr)
{
    this->undoActionPtr = undoActionPtr;
}


QAction* MasterActions::getUndoActionPtr()
{
    return undoActionPtr;
}


void MasterActions::setRedoActionPtr(QAction* redoActionPtr)
{
    this->redoActionPtr = redoActionPtr;
}


QAction* MasterActions::getRedoActionPtr()
{
    return redoActionPtr;
}


void MasterActions::setSetProjectConfigActionPtr(QAction* setProjectConfigActionPtr)
{
    this->setProjectConfigActionPtr = setProjectConfigActionPtr;
}


QAction* MasterActions::getSetProjectConfigActionPtr()
{
    return setProjectConfigActionPtr;
}


void MasterActions::setBuildActionPtr(QAction* buildActionPtr)
{
    this->buildActionPtr = buildActionPtr;
}


QAction* MasterActions::getBuildActionPtr()
{
    return buildActionPtr; 
}


void MasterActions::setCleanAndBuildActionPtr(QAction* cleanAndBuildActionPtr)
{
    this->cleanAndBuildActionPtr = cleanAndBuildActionPtr;
}


QAction* MasterActions::getCleanAndBuildActionPtr()
{
    return cleanAndBuildActionPtr;
}


void MasterActions::setConfigForRunActionPtr(QAction* configForRunActionPtr)
{
    this->configForRunActionPtr = configForRunActionPtr;
}


QAction* MasterActions::getConfigForRunActionPtr()
{
    return configForRunActionPtr;
}


void MasterActions::setRunActionPtr(QAction* runActionPtr)
{
    this->runActionPtr = runActionPtr;
}


QAction* MasterActions::getRunActionPtr()
{
    return runActionPtr;
}


void MasterActions::setDebugActionPtr(QAction* degugActionPtr)
{
    this->debugActionPtr = debugActionPtr;
}


QAction* MasterActions::getDebugActionPtr()
{
    return debugActionPtr;
}


void MasterActions::setGitActionPtr(QAction* gitActionPtr)
{
    this->gitActionPtr = gitActionPtr;
}


QAction* MasterActions::getGitActionPtr()
{
    return gitActionPtr;
}


// South
void MasterActions::setOutputSouthActionPtr(QAction* outputSouthActionPtr)
{
    this->outputSouthActionPtr = outputSouthActionPtr;
}


QAction* MasterActions::getOutputSouthActionPtr()
{
    return outputSouthActionPtr;
}


void MasterActions::setTerminalSouthActionPtr(QAction* terminalSouthActionPtr)
{
    this->terminalSouthActionPtr = terminalSouthActionPtr;
}


QAction* MasterActions::getTerminalSouthActionPtr()
{
    return terminalSouthActionPtr;
}


// East
void MasterActions::setNavEastActionPtr(QAction* navWestActionPtr)
{
    this->navEastActionPtr = navWestActionPtr;
}


QAction* MasterActions::getNavEastActionPtr()
{
    return navEastActionPtr;
}


// West
void MasterActions::setProjectWestActionPtr(QAction* projectWestActionPtr)
{
    this->projectWestActionPtr = projectWestActionPtr;
}


QAction* MasterActions::getProjectWestActionPtr()
{
    return projectWestActionPtr;
}


template<class X>
void MasterActions::connectToNewFileAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleNewFileActionSlot() ) );
}


template<class X>
void MasterActions::connectToNewTerminalAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleNewTerminalActionSlot() ) );
}


template<class X>
void MasterActions::connectToNewProjectAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleNewProjectActionSlot() ) );
}


template<class X>
void MasterActions::connectToOpenProjectAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleOpenProjectActionSlot() ) );
}


template<class X>
void MasterActions::connectToSaveAllAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleSaveAllActionSlot() ) );
}


template<class X>
void MasterActions::connectToUndoAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleUndoActionSlot() ) );
}


template<class X>
void MasterActions::connectToRedoAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleRedoActionSlot() ) );
}


template<class X>
void MasterActions::connectToSetProjectConfigAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleSetProjectConfigActionSlot() ) );
}


template<class X>
void MasterActions::connectToBuildAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleBuildActionSlot() ) );
}


template<class X>
void MasterActions::connectToCleanAndBuildAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleCleanAndBuildActionSlot() ) );
}


template<class X>
void MasterActions::connectToConfigForRunAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleConfigForRunActionSlot() ) );
}


template<class X>
void MasterActions::connectToRunAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleRunActionSlot() ) );
}


template<class X>
void MasterActions::connectToDebugAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleDebugActionSlot() ) );
}


template<class X>
void MasterActions::connectToGitAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleGitActionSlot() ) );
}


// South
template<class X>
void MasterActions::connectToOutputSouthAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleOutputSouthActionSlot() ) );
}


template<class X>
void MasterActions::connectToTerminalSouthAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleTerminalSouthActionSlot() ) );
}


// East
template<class X>
void MasterActions::connectToNavEastAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleNavEastActionSlot() ) );
}


// West
template<class X>
void MasterActions::connectToProjectWestAction(X* component)
{
    connect(component, SIGNAL(released() ), this, SLOT(handleProjectWestActionSlot() ) );
}


QString* MasterActions::toString()
{
    QString* tmp = new QString();
    tmp->append("***method stub***");
    
    return tmp;
}


MasterActions::~MasterActions()
{
    ;
}